kview/src/main/java/dev/ikm/komet/kview/mvvm/view/journal/JournalController.java

```diff
@@ -1391,21 +1442,25 @@ private void makeNewPatternWindow(ObservableViewNoOverride windowView, NidTextEn
         stampViewModel.setPropertyValue(PATHS_PROPERTY, stampViewModel.findAllPaths(viewProperties), true)
                 .setPropertyValue(MODULES_PROPERTY, stampViewModel.findAllModules(viewProperties), true);
 
+        StateMachine patternSM = StateMachine.create(new PatternDetailsPattern());
+
+        ValidationViewModel patternViewModel = new PatternViewModel()
+                .setPropertyValue(VIEW_PROPERTIES, viewProperties)
+                .setPropertyValue(MODE, CREATE)
+                .setPropertyValue(STAMP_VIEW_MODEL, stampViewModel)
+                .setPropertyValue(PATTERN_TOPIC, UUID.randomUUID())
+                .setPropertyValue(STATE_MACHINE, patternSM);
```

Comment:
**Suggestion:**
In another ticket, we could improve how we are using the state machine we can do the following:

- Don't pass the state machine in the ViewModel. Leave it inside of the `PropertiesController`. Events and subscribers can listen and make transitions. Any state from the `PatternDetailsController` such as property toggle open or closed. 
- Use the state machine's `.when()` methods. Here you can perform actions when a state is encountered. https://github.com/carldea/axonic/wiki#when-code-block-executions